handlers/search.js

```diff
@@ -91,28 +95,36 @@ function searchPage(request, reply) {
   const filters = getFilters(query);
   const advancedSearch = query.advanced_search || (Object.keys(filters).length > 0);
 
-  trials.search(queryStr, page, perPage, filters).then((_trials) => {
-    const currentPage = page || 1;
-    const pagination = generatePaginationLinks(
-      request.url, currentPage,
-      perPage, maxPages,
-      _trials.total_count
-    );
-    const queryDescription = queryStr || 'clinical trial data';
-    const title = `Search results for ${queryDescription} in OpenTrials`;
-    const description = `OpenTrials has ${_trials.total_count} matches for
+  sources.list().then((_sources) => {
+    trials.search(queryStr, page, perPage, filters).then((_trials) => {
```

Comment:
You could also extract the inner function `(_sources, _trials) => {}` to a private function. The complicator is that it requires the `request` and `reply` passed to this function. You have two options then: creating a function that returns another function, or defining the function inside the scope of this function, so it can access its parameters.

```javascript
// 1st option
function _doStuff(request, reply) {
  return function _foobar(_sources, _trials) {
    // ..
  }
}

function searchPage(request, reply) {
  return Promise.join(
    sources.list(),
    trials.search( /* ... */ ),
    _doStuff(request, reply)
  );
}

// 2nd option
function searchPage(request, reply) {
  function _fooBar(_sources, _trials) {
    // ...
  }

  return Promise.join(
    // ...
  );
}
```

I don't have a strong preference either way, or even on keeping the function as is.